api/v1/backupstoragelocation_types.go

```diff
@@ -21,33 +21,85 @@ import (
 	"k8s.io/apimachinery/pkg/types"
 )
 
-// +genclient
-// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+func init() {
+	SchemeBuilder.Register(&BackupStorageLocation{}, &BackupStorageLocation{})
+}
 
-// BackupStorageLocation is a location where Velero stores backup objects.
-type BackupStorageLocation struct {
-	metav1.TypeMeta `json:",inline"`
+// BackupStorageLocationSpec defines the desired state of a Velero BackupStorageLocation
+type BackupStorageLocationSpec struct {
+	// Provider is the provider of the backup storage.
+	Provider string `json:"provider"`
 
+	// Config is for provider-specific configuration fields.
 	// +optional
-	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Config map[string]string `json:"config,omitempty"`
+
+	StorageType `json:",inline"`
+
+	// AccessMode defines the permissions for the backup storage location.
+	// +optional
+	AccessMode BackupStorageLocationAccessMode `json:"accessMode,omitempty"`
+
+	// BackupSyncPeriod defines how frequently to sync backup API objects from object storage. A value of 0 disables sync.
+	// +optional
+	// +nullable
+	BackupSyncPeriod *metav1.Duration `json:"backupSyncPeriod,omitempty"`
+}
+
+// BackupStorageLocationStatus defines the observed state of BackupStorageLocation
+type BackupStorageLocationStatus struct {
+	// Phase is the current state of the BackupStorageLocation.
+	// +optional
+	Phase BackupStorageLocationPhase `json:"phase,omitempty"`
 
+	// LastSyncedTime is the last time the contents of the location were synced into
+	// the cluster.
 	// +optional
-	Spec BackupStorageLocationSpec `json:"spec,omitempty"`
+	// +nullable
+	LastSyncedTime *metav1.Time `json:"lastSyncedTime,omitempty"`
 
+	// LastSyncedRevision is the value of the `metadata/revision` file in the backup
+	// storage location the last time the BSL's contents were synced into the cluster.
+	//
+	// Deprecated: this field is no longer updated or used for detecting changes to
+	// the location's contents and will be removed entirely in v2.0.
 	// +optional
+	LastSyncedRevision types.UID `json:"lastSyncedRevision,omitempty"`
+
+	// AccessMode is an unused field.
+	//
+	// Deprecated: there is now an AccessMode field on the Spec and this field
+	// will be removed entirely as of v2.0.
+	// +optional
+	AccessMode BackupStorageLocationAccessMode `json:"accessMode,omitempty"`
+}
+
+// +kubebuilder:object:root=true
```

Comment:
should we also add the annotation to generate RBAC?

Reply:
@ashish-amarnath I really don't know. The original BSL didn't have that marker. Please let me know if you think I should add it, but also if you could help me understand why it'd be great.

Reply:
KubeBuilder is able to generate RBAC for the controllers to run wit, based on the annotation. The question was whether we want to use that. We can table that discussion for when we move all the controllers.